runtime/common/src/impls.rs

```diff
@@ -163,3 +165,90 @@ impl From<u128> for InstanceId {
 		Self(v)
 	}
 }
+
+impl Default for PermissionRoles {
+	fn default() -> Self {
+		Self {
+			admin: AdminRoles::empty(),
+			tranche_investor: TrancheInvestors::empty(),
```

Comment:
Tranche investors are the memebers that can hold restricted tokens with specific valid time. This cannot be part of this Role as this will ideally come from restricted token pallet. 

Pool pallet will check if the at the time of supply if the the investor is member that can hold the restricted tokens.
cc: @offerijns

Reply:
With memeberlist, there will be minumum valid time for the member per pool/tranche. That Minumum valid time ideally be present in the  restricted tokens. When a MemebrlistAdmin tries to add a new tranche investor or update the existing tranche investor, they shoukd ideally add that through the restricted tokens pallet. 

If the permission pallet role contains the tranche investor, then it should ideally get the MinumumValid time for that tranche in a pool from Resricted tokens. So this means, permissions pallets should know what a Restricted token is. I feel that defeats the purpose of the pallet being generic

Reply:
But this would be possible. Either we adapt the TrancheInvestor to be of type 
```rust
TrancheInvestor(u32, Since)
```
where since simply stores the moment in time or a block number when this was added.

Or we store the moment in time, when the restricted-token pallets adds an tranche investor in the pallet-restricted-tokens and it can check it there.

Reply:
Ah, we would need to go with the second approach, as the actual type in storage is never exposed in the trait interface.

Reply:
The requirements for the restricted tokens are 
- to check if the member can hold the tokens at this instant
- Ability to add or modify a member with a minimum holding time by a Memberlist Admin
- (Good to have) to be able to update minimum holding time for each pool/tranche instead of default 7 days


> Can't you use the Permissions trait in the restricted tokens pallet and add/remove users there?

Yes but there would be two extrinsics that does the same. One in Permissions and other in restricted tokens. If we decide to not have that in restricted tokens, then permissions pallets should support variable admin origins where only a Memberlist admin can add/update the members, and also further fetches the minimum delay for the given tranche.

Reply:
The pallet-permissions allows root basically to add and remove any permission. Which I think is fine. And users can purge their permissions if wanted. 

Other pallets, that need additional checks, should do themselves. At least this is my opinion.

Reply:
A pool owner should be able to update the admin level permissions for the given pool. Making this a democracy process is not required since its always up to the pool owner or manager.

> And users can purge their permissions if wanted.

This is not always the case 😉 

> Other pallets, that need additional checks, should do themselves. At least this is my opinion.

True but the permissions pallet should ideally not provide the these extrinsics by itself then. It should only act like a storage layer and provide internal calls for the pallets. 
Then pool will ideally have the extrinsincs that allow its pool owner to make these changes
and restricted tokens will be able to make changes after verifying the membership admin role.

Reply:
So restricted tokens could do like: 

```rust
fn OnMinHoldingEmpty<T:Config> -> T::MinHolding { T::MinHolding::get() }

pub type MinHolding<T: Config> = StorageDoubleMap<
		_,
		Blake2_128Concat,
		T::PoolId,
		Blake2_128Concat,
		T::TrancheId,
		T::MinHolding,
                ValueQuery,
                OnMinHoldingEmpty
	>;
```

for the holding period and update this on a pool-tranche-basis if needed or use the default value.

Reply:
Or are you worried that in another place of the codebase we change the permissions of `TrancheInvestor`s?

Reply:
We just need to doublemap (pool and tranche) to a moment on the restricted tokens. This will be checked along with MemberlistAdmin role before adding/removing permisssions on the permissions pallet

> Or are you worried that in another place of the codebase we change the permissions of TrancheInvestors?

yes, there should be no other way this permissions will be changed through externisics apart from restricted tokens